spec/operators/delay-spec.js

```diff
@@ -1,11 +1,79 @@
 /* globals describe, it, expect, expectObservable, hot, rxTestScheduler */
 var Rx = require('../../dist/cjs/Rx');
+var Observable = Rx.Observable;
 
 describe('Observable.prototype.delay()', function () {
   it('should delay by specified timeframe', function () {
-    var source = hot('--a--|');
-    var expected =   '-----a--|';
+    var e1 =   hot('--a--b--|');
+    var expected = '-----a--b--|';
 
-    expectObservable(source.delay(30, rxTestScheduler)).toBe(expected);
+    expectObservable(e1.delay(30, rxTestScheduler)).toBe(expected);
+  });
+
+  it('should delay by absolute time period', function () {
+    var e1 =   hot('--a--b--|');
+    var expected = '-----a--b--|';
+    var absoluteDelay = new Date(rxTestScheduler.now() + 30);
+
+    expectObservable(e1.delay(absoluteDelay, rxTestScheduler)).toBe(expected);
+  });
+
+  it('should delay by absolute time period after subscription', function () {
+    var e1 =   hot('---^--a--b--|');
+    var expected =    '------a--b--|';
+    var absoluteDelay = new Date(rxTestScheduler.now() + 30);
+
+    expectObservable(e1.delay(absoluteDelay, rxTestScheduler)).toBe(expected);
+  });
+
+  it('should raise error when source raises error by specified timeframe', function () {
+    var e1 =   hot('---a---b---#');
+    var expected = '------a---b---#';
+
+    expectObservable(e1.delay(30, rxTestScheduler)).toBe(expected);
+  });
+
+  it('should raise error when source raises error by absolute time period', function () {
+    var e1 =   hot('--a--b--#');
+    var expected = '-----a--b--#';
+    var absoluteDelay = new Date(rxTestScheduler.now() + 30);
+
+    expectObservable(e1.delay(absoluteDelay, rxTestScheduler)).toBe(expected);
+  });
+
+  it('should raise error when source raises error by absolute time period after subscription', function () {
+    var e1 =   hot('---^---a---b---#');
+    var expected =    '-------a---b---#';
+    var absoluteDelay = new Date(rxTestScheduler.now() + 30);
```

Comment:
This behavior is bit different to RxJS4 [test case] (https://github.com/Reactive-Extensions/RxJS/blob/master/tests/observable/delay.js#L156-L256), in RxJS4 test case is described as

```
source  = '---a---b---#'
delayed = '----a----b-#'; 
delayed = hot(source).delay(10);
```

that `error` is emitted as soon as source emits. Per [documantation](https://github.com/Reactive-Extensions/RxJS/blob/master/doc/api/core/operators/delay.md), behavior is defined as 

`Time shifts the observable sequence by dueTime. The relative time intervals between the values are preserved.`
think `error` should be considered as same as normal value element, but maybe that's something I understood behavior incorrectly(or either read RxJS4 test cases incorrectly?), asking confirmation. @blesh @staltz